config/config.go

```diff
@@ -142,6 +142,9 @@ type GlobalConfig struct {
 	TimeoutOffset model.Duration `yaml:"scrape_timeout_offset"` // offset to subtract from timeout in seconds
 	MaxConns      int            `yaml:"max_connections"`       // maximum number of open connections to any one target
 	MaxIdleConns  int            `yaml:"max_idle_connections"`  // maximum number of idle connections to any one target
+	MaxLabelNameLen int          `yaml:"max_label_name_len"`    // maximum length of label name
+	MaxLabelValueLen int         `yaml:"max_label_value_len"`   // maximum length of label value
+	MaxJsonLabels int            `yaml:"max_json_labels"`       // maximum number of labels extracted from json column
```

Comment:
I would drop these altogether. If you can't rely on your input data, you've got bigger problems than how to truncate it.

In particular, Golang maps have randomized order (beyond your usual hash map, that would have the same order as long as you used the same set of keys), so truncating the map to the first 10 entries would give you 10 random labels. That is very likely not what anyone wants. (Plus, the Prometheus client library will not accept metrics whose labels vary from one instance to the next.)

Reply:
In monitoring SQL you can not always rely on you input data, because well database is usually not managed by the same people as monitoring and you usually get only some restricted views, whose creation is out of your competence. I know, you can restrict values in SQL query, but lets be honest - who does it ;-)